MKon Authenticated Data (#190)
monero-ecosystem/outreach-docs: monero-outreach-docs/en/transcriptions/monero-konferenco-2019/MoneroKon_2019_Authenticated_Data.md

﻿thank you again let's give Bennet Ben

fish along with joint work with Benedict

buns and Dan Bona a hand for

authenticated data structures for

blockchain consensus with minimal

storage thanks Brandon

okay so

at Green

got it no that's green arrow perfect so

okay

block genes are transactional databases

so naturally as with any transactional

database they grow in size with use but

the significant thing about you know

> block chains like Bitcoin or Manero is

Reviewer: ```suggestion
block chains like Bitcoin or Monero is

```

previous transaction spent so that

transaction should be rejected but this

miner with no storage has no way of

knowing this so what we would like from

stateless verification

the ability of a miner with very very

very small compact storage like just

storing a little hash representation of

everything that's happened so far would

be able to you know listen to

transactions coming in and say up this

should be rejected or up this should be

accepted so how could we possibly do

that how could we possibly define these

compact hashes in a way that you know

all the nodes can just store this hash

and still be able to check the validity

of transactions well clearly we need

something additional okay some kind of

additional proof that will be sent along

with transactions so we'll make the

transaction slightly more complicated

but what I'm going to talk about as well

how is that proof generated how

efficient is it and who generates the

proof first of all but to take a step

back just from maybe a philosophical

perspective or you know a more abstract

view what we're calling for is a

separation between the consensus layer

of the blockchain and this the state

> storage of the box chain and both of

Reviewer: ```suggestion
storage of the blockchain and both of

```

these we want to be decentralized or

distributed and fault tolerant but we

can really look at them as two different

systems and one which aims to you know

maintain the and store all the data in

the blockchain and and be able to

provide basically proofs and state

updates to the consensus which is really

not focused on on necessarily providing

you know a really high performance

database but rather just being able to

participate in agreement and and verify

that things look good so let's focus on

a concrete now it's like specific

example of a blockchain model and so

> I'll focus on UT X OS as since it's

Reviewer: ```suggestion
I'll focus on UTXOs as since it's

```

> relevant to both Bitcoin and manera so

Reviewer: ```suggestion
relevant to both Bitcoin and Monero so

```

> what are you TX O's it's in case you

Reviewer: ```suggestion
what are your TXO's it's in case you

```

know just as a reminder whatever every

transaction consumes inputs think of

like coins assigned addresses that are

the inputs and transactions every

transaction creates outputs

coins are reassigned to new addresses

> and UTX those are simply the transaction

Reviewer: ```suggestion
and UTXOs those are simply the transaction

```
